applications/tari_merge_mining_proxy/src/main.rs

```diff
@@ -0,0 +1,704 @@
+                                match get_monero_data(&json, transient.clone().monero_seed.unwrap()) {
+                                    Ok(pow_data) => {
+                                        let mut tariheader: grpc::BlockHeader =
+                                            block.header.ok_or(MmProxyError::MissingDataError(format!("Header")))?;
+                                        let mut powdata = tariheader
+                                            .pow
+                                            .clone()
+                                            .ok_or(MmProxyError::MissingDataError(format!("Pow data")))?;
+                                        powdata.pow_data = bincode::serialize(&pow_data).map_err(|_| {
+                                            MmProxyError::ParseError(format!("Failure to serialize block "))
+                                        })?;
+                                        tariheader.pow = Some(powdata);
+                                        block.header = Some(tariheader);
+                                        trace!(target: LOG_TARGET, "Tari Block {:?}", block);
+                                        match rt.block_on(grpcclient.submit_block(block)) {
+                                            Ok(result) => {
+                                                result.into_inner();
+                                            },
+                                            Err(e) => {
+                                                debug!(target: LOG_TARGET, "{:#}", e);
+                                            },
+                                        }
+                                    },
+                                    Err(e) => {
+                                        debug!(target: LOG_TARGET, "{:#}", e);
+                                    },
+                                }
+                            }
+                        } else {
+                            debug!(target: LOG_TARGET, "{:#}", response);
+                        }
+                    },
+                    Err(e) => {
+                        debug!(target: LOG_TARGET, "{:#}", e);
+                    },
+                },
+                "getblocktemplate" => {
+                    // Add merge mining tag on blocktemplate request
+                    // PowAlgo = 0 is Monero
+                    match grpcclientresult {
+                        Ok(mut grpcclient) => {
+                            match rt.block_on(grpcclient.get_new_block_template(grpc::PowAlgo { pow_algo: 0 })) {
+                                Ok(new_block_template_response) => {
+                                    let new_template = new_block_template_response
+                                        .into_inner()
+                                        .new_block_template
+                                        .ok_or(MmProxyError::MissingDataError(format!("New block template")))?;
```

Comment:
Clippy will produce lots of warnings for the `ok_or` and `format!`, there are multiple instances of these.

```
.ok_or_else(|| MmProxyError::MissingDataError("New block template".to_string())?;
```